code/game/atoms_movable.dm

```diff
@@ -655,64 +638,17 @@
 */
 
 /atom/movable/proc/set_stationloving(state, inform_admins=FALSE)
-	var/currently = (flags_2 & STATIONLOVING_2)
-
-	if(inform_admins)
-		flags_2 |= INFORM_ADMINS_ON_RELOCATE_2
-	else
-		flags_2 &= ~INFORM_ADMINS_ON_RELOCATE_2
-
-	if(state == currently)
-		return
-	else if(!state)
-		STOP_PROCESSING(SSinbounds, src)
-		flags_2 &= ~STATIONLOVING_2
-	else
-		START_PROCESSING(SSinbounds, src)
-		flags_2 |= STATIONLOVING_2
-
-/atom/movable/proc/relocate()
-	var/targetturf = find_safe_turf()
-	if(!targetturf)
-		if(GLOB.blobstart.len > 0)
-			targetturf = get_turf(pick(GLOB.blobstart))
-		else
-			throw EXCEPTION("Unable to find a blobstart landmark")
 
-	if(ismob(loc))
-		var/mob/M = loc
-		M.transferItemToLoc(src, targetturf, TRUE)	//nodrops disks when?
-	else if(loc.SendSignal(COMSIG_CONTAINS_STORAGE))
-		loc.SendSignal(COMSIG_TRY_STORAGE_TAKE, src, targetturf, TRUE)
-	else
-		forceMove(targetturf)
-	// move the disc, so ghosts remain orbiting it even if it's "destroyed"
-	return targetturf
+	var/datum/component/stationloving/component = GetComponent(/datum/component/stationloving)
 
-/atom/movable/proc/check_in_bounds()
-	if(in_bounds())
+	if(state)
+		if (!component)
+			AddComponent(/datum/component/stationloving, inform_admins)
+		else if (component.inform_admins != inform_admins)
+			AddComponent(/datum/component/stationloving, inform_admins)

```

Comment:
You're replicating work done by AddComponent here, just remove this proc and make the component use `COMPONENT_DUPE_UNIQUE_PASSARGS` to detect when inform_admins has changed